api/nakadi-event-bus-api.yaml

```diff
@@ -0,0 +1,1207 @@
+      parameters:
+        - name: X-Flow-Id
+          in: header
+          description: |
+            The flow id of the request, which is written into the logs and passed to called services. Helpful
+            for operational troubleshooting and log analysis.
+          type: string
+          format: flow-id
+      responses:
+        '200':
+          description: Ok
+          schema:
+            type: array
+            items:
+              $ref: '#/definitions/EventType'
+        '401':
+          description: Client is not authenticated
+          schema:
+            $ref: '#/definitions/Problem'
+        '500':
+          description: Server error
+          schema:
+            $ref: '#/definitions/Problem'
+
+    post:
+      tags:
+        - schema-registry-api
+      description: |
+        Creates a new `EventType`.
+
+        **Implementation note:** The creation of an EventType implicitly creates the structures in the
+        backing messaging implementation needed for the reception and persistency of the Events. Considering
+        that at this time only Kafka is used, this corresponds to the creation of a Topic. If so desired,
+        clients can interact directly with the topic using the low level API (for existing restrictions see
+        the corresponding methods on the topic-api).
+
+        The fields validation-strategies, enrichment-strategies and partition-resolution-strategy have all an
+        effect on the incoming Event of this EventType. For its impacts on the reception of events please
+        consult the Event submission API methods.
+
+        * Validation strategies define an array of `EventValidationStrategy` to be evaluated on reception of
+        an `Event` of this `EventType`. Details of usage can be found in an external document (TBD link to
+        document).
+        **Notes**: Some EventValidationStrategies might support additional parameters. The behaviour is
+        specific to each strategy and one is suggested to consult its `doc` field for detailes of operation.
+        The default `SchemaValidationStrategy` supports passing fields to be ignored if the qualifier matches
+        the input.
+
+        * TDB Enrichment strategy
+
+        * TDB Partition resolution strategy
+
+        * The schema of an `EventType` is defined as an `EventTypeSchema`. Currently only json-schema is
+        supported.
+
+        Following violations are enforced. Not meeting them will fail the request with a 422 Unprocessable
+        entity (details are provided in the Problem object):
+
+        * Event name on creation not begin unique.
+
+        * Using `EventTypeSchema.type` other than json-schema or passing a `EventTypeSchema.schema`
+        that is invalid with respect to the schema's type.
+
+        * Referring any of Validation, Enrichment or Partitioning strategies that does not exist or whose
+        parameterization is deemed invalid.
+
```

Comment:
@igoryankov you are right. Rebalancing is on the roadmap. But this will come only after the Subscription API is there, that will make it possible to repartition using the same protocol, as during the rebalancing of partitions when new clients are added to the subscription.